### ISSUE TITLE
System.TypeInitializationException on Linux

### ISSUE THREAD

**Issue by brwyatt** [BattletechModders/ModTek#132]
## Summary
Following [the instructions](https://github.com/BattletechModders/ModTek/wiki/The-Drop-Dead-Simple-Guide-to-Installing-BTML-&-ModTek-&-ModTek-mods) to install both 0.6.1 and 0.7.0 both fail with `System.TypeInitializationException: The type initializer for 'ModTekInjector.Injector' threw an exception.`.

I see a few other issues for running on Linux that are different than this one that others have resolved, so I assume the injector is working for them. I'm wondering if this might be something specific to my version of Ubuntu (18.04) that isn't present on the previous LTS (16.04).

## Fresh install log
```Bash
brwyatt@crimson:BATTLETECH$ pwd
/home/brwyatt/.local/share/Steam/steamapps/common/BATTLETECH
brwyatt@crimson:BATTLETECH$ mkdir Mods
brwyatt@crimson:BATTLETECH$ cd Mods/
brwyatt@crimson:Mods$ unzip ~/Downloads/ModTek-0.7.0.zip 
Archive:  /home/brwyatt/Downloads/ModTek-0.7.0.zip
   creating: ModTek/
  inflating: ModTek/0Harmony.dll     
  inflating: ModTek/Ionic.Zip.dll    
  inflating: ModTek/ModTek.dll       
 extracting: ModTek/modtekassetbundle  
  inflating: ModTek/ModTekInjector.exe  
  inflating: ModTek/Mono.Cecil.dll   
  inflating: ModTek/Mono.Cecil.Mdb.dll  
  inflating: ModTek/Mono.Cecil.Pdb.dll  
  inflating: ModTek/Mono.Cecil.Rocks.dll  
  inflating: ModTek/Newtonsoft.Json.dll  
  inflating: ModTek/System.Runtime.Serialization.dll  
brwyatt@crimson:Mods$ ls
ModTek
brwyatt@crimson:Mods$ cd ModTek/
brwyatt@crimson:ModTek$ mono ModTekInjector.exe 
WARNING: The runtime version supported by this application is unavailable.
Using default runtime: v4.0.30319

Unhandled Exception:
System.TypeInitializationException: The type initializer for 'ModTekInjector.Injector' threw an exception.
brwyatt@crimson:ModTek$ 
```

## System Information
```Bash
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.2 LTS
Release:	18.04
Codename:	bionic
$ mono --version
Mono JIT compiler version 4.6.2 (Debian 4.6.2.7+dfsg-1ubuntu1)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           __thread
	SIGSEGV:       altstack
	Notifications: epoll
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	LLVM:          supported, not enabled.
	GC:            sgen
```
## Additional information
I see some notes in the 0.6.1 release indicating some fixes for Linux. Are there any additional install steps or dependencies needed to run on Linux?

Let me know if you need additional information.

**Comment by Mpstark** [comment]
This issue crept up on reddit and was the reporter said:

> OK folks, here's what fixed it:
> export TERM=xterm
> After that, the injector worked just fine. Note to all this is on a stock version of Ubuntu 18.10.

I do not know why this is necessary now.

**Comment by brwyatt** [comment]
Unfortunately, this doesn't seem to fix it for me, I still get the same exception.

I'd actually come across another issue that had mentioned the XTERM (and Locale/LANG) issue, and tried it with no results, unfortunately.

```
brwyatt@crimson:ModTek$ export TERM=xterm
brwyatt@crimson:ModTek$ echo $LANG
en_US.UTF-8
brwyatt@crimson:ModTek$ echo $TERM
xterm
brwyatt@crimson:ModTek$ mono ModTekInjector.exe 
WARNING: The runtime version supported by this application is unavailable.
Using default runtime: v4.0.30319

Unhandled Exception:
System.TypeInitializationException: The type initializer for 'ModTekInjector.Injector' threw an exception.
```

**Comment by brwyatt** [comment]
I went and dug up the [Reddit thread](https://www.reddit.com/r/BattleTechMods/comments/atxl6b/linux_modtek_issues/) mentioned. While `export TERM=xterm` didn't fix things, installing `mono-complete` did seem to fix it (also with the TERM fix). So it now at least injects into `Assembly-CSharp.dll`. However, game start crashes with `ModTek Died!`. The ModTek.log contains the following:

```
Uncaught ModTek exception!
System.ArgumentNullException: Argument cannot be null.
Parameter name: source
  at System.Linq.Check.SourceAndPredicate (System.Object source, System.Object predicate) [0x00000] in <filename unknown>:0 
  at System.Linq.Enumerable.Where[String] (IEnumerable`1 source, System.Func`2 predicate) [0x00000] in <filename unknown>:0 
  at ModTek.ModTek.CallFinishedLoadMethods () [0x00000] in <filename unknown>:0 
  at ModTek.ModTek+<FinishLoop>d__125.MoveNext () [0x00000] in <filename unknown>:0 
  at ModTek.UI.ProgressPanel+ProgressBarLoadingBehavior+<RunWorkList>d__20.MoveNext () [0x00000] in <filename unknown>:0 
```

I'll note that I have no mods installed or anything at this point, I've only run the injector. Running the restore command also seems to work and makes the game playable again.

**Comment by brwyatt** [comment]
Update: interestingly, I'm only getting the `ModTek Died!` error with 0.7.0. 0.6.1 appears to work fine when injected.

Since 0.7.0 says if you injected with 0.6.x, you don't need to re-inject, I'm just going to unzip 0.7.0 over top of 0.6.1, but it seems kinda strange that 0.7.0 would fail the inject in this way when 0.6.1 works.

Let me know if you want/need any further testing/data to help sort out what's breaking in the 0.7.0 injector.

**Comment by Mpstark** [comment]
@brwyatt Yep, looks like there is a bug where if there are no loaded mods, it dies. I've made this mistake before too, ugh. Here's a fixed version -- the fix will get rolled into 0.7.1

[ModTek.zip](https://github.com/BattletechModders/ModTek/files/2904719/ModTek.zip)

**Comment by Mpstark** [comment]
0.7.1 should fix the NRE for no loaded mods.
